### PR TITLE
microsite a11y fix

### LINKED ISSUE #21882: [Accessibility] Microsite - missing Alt text on multiple images
### 📜 Description

Missing on some pages (Docs); heading text repeated in alt on others (Plug-ins).

This helps those who use screen readers or are on slower connections. Alt text should be defined based on image type and the [w3C Alt decision tree](https://www.w3.org/WAI/tutorials/images/decision-tree/).

### 👍 Expected behavior

Images to have alternate text. [Learn more about the alt attribute](https://dequeuniversity.com/rules/axe/4.8/image-alt).

### 👎 Actual Behavior with Screenshots

Example: 
Screenshot 2023-12-15 at 13 12 23

### 👟 Reproduction steps

1. Go to https://backstage.io/
2. [Run the Lighthouse browser extension ](https://chromewebstore.google.com/detail/lighthouse/blipmdconlkpinefehnmjammfjpmpbjk?pli=1)

Alternatively:
1. Run the microsite locally
2. [Run the Lighthouse CLI locally when developing new features](https://backstage.io/docs/accessibility/#run-the-lighthouse-cli-locally-when-developing-new-features)

### 👀 Have you spent some time to check if this bug has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/backstage/backstage/blob/master/CODE_OF_CONDUCT.md)

### Are you willing to submit PR?

None

### LINKED ISSUE #21883: [Accessibility] Microsite - missing link titles
### 📜 Description

Missing on Home page

Link title is crucial, either as visible text or an accessible label to satisfy [2.4.4 Link Purpose](https://www.w3.org/WAI/WCAG22/Understanding/link-purpose-in-context.html)

### 👍 Expected behavior

[Links to have discernible text](https://dequeuniversity.com/rules/axe/4.8/link-name)

### 👎 Actual Behavior with Screenshots

Example: 
Screenshot 2023-12-15 at 13 22 44

### 👟 Reproduction steps

1. Go to https://backstage.io/
2. [Run the Lighthouse browser extension ](https://chromewebstore.google.com/detail/lighthouse/blipmdconlkpinefehnmjammfjpmpbjk?pli=1)

Alternatively:
1. Run the microsite locally
2. [Run the Lighthouse CLI locally when developing new features](https://backstage.io/docs/accessibility/#run-the-lighthouse-cli-locally-when-developing-new-features)

_No response_

### 👀 Have you spent some time to check if this bug has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/backstage/backstage/blob/master/CODE_OF_CONDUCT.md)

### Are you willing to submit PR?

None

### LINKED ISSUE #21885: [Accessibility] Microsite - Color Contrast fails to meet WCAG conformance
### 📜 Description

Example: Selection drop-downs, “are you using Backstage”, privacy info, links in docs text

Users with low vision may find controls inoperable or be unable to read on-page text

Some of these elements returned contrasts of 1.27:1 (such as the privacy paragraph) thus new color pallets may need to be chosen by design

### 👍 Expected behavior

Color contrast succeeds. One can use [Contrast Checker](https://webaim.org/resources/contrastchecker/) to verify this. 

### 👎 Actual Behavior with Screenshots

(couldn't find an example of the issue of when the audit was first conducted but here is another one)

Example of links:
Screenshot 2023-12-15 at 14 02 43

### 👟 Reproduction steps

1. Go to https://backstage.io/
2. [Run the Lighthouse browser extension ](https://chromewebstore.google.com/detail/lighthouse/blipmdconlkpinefehnmjammfjpmpbjk?pli=1)

Alternatively:
1. Run the microsite locally
2. [Run the Lighthouse CLI locally when developing new features](https://backstage.io/docs/accessibility/#run-the-lighthouse-cli-locally-when-developing-new-features)

### 👀 Have you spent some time to check if this bug has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/backstage/backstage/blob/master/CODE_OF_CONDUCT.md)

### Are you willing to submit PR?

None

### PR DESCRIPTION
**Pull request by npiyush97**
## Hey, I just made a Pull Request!

closes #21883, closes #21882,closes #21885
#### :heavy_check_mark: Checklist

- [ ] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [ ] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Review by awanlin — APPROVED** [review]
Thanks for the corrections @npiyush97 🚀

**Comment by npiyush97** [comment]
@emmaindal div or text in anchor tag?

**Comment by npiyush97** [comment]
@benjdlambert I don't know who is responsible for to docs so I am asking you I'm adding underline on anchor tag fixes #21885 because the underline differentiates the link from surrounding text.

**Comment by npiyush97** [comment]
> > separate question for maintainers why prettierignore microsite?
> 
> The microsite is a bit unique and isn't like working with the packages/plugins that make up Backstage itself. It has it's own settings for this which is why it's ignored at the root. Hope that helps explain?

Just because yarn prettier:fix wont work on microsite 
I have to comment uncomment in prettier ignore to fix format.
Or there is another way i dont know.

**Comment by awanlin** [comment]
Ah, yeah, that's not very convenient @npiyush97 

Could you add it as part of this PR here then: https://github.com/backstage/backstage/blob/9667fc08408acdb6e8ad34b1d1443c1b2b3dbb97/microsite/package.json#L9

File: microsite/src/theme/customTheme.scss
Comment (Rugvip) [inline microsite/src/theme/customTheme.scss:9]: I don't think this is the fix we want here tbh. It doesn't make the text itself more readable, which is still important. Perhaps we could separate this one out to a separate PR so we can fix the rest quicker?

**Review by Rugvip — CHANGES_REQUESTED** [review]
Thank you! 👍

File: microsite/src/theme/customTheme.scss
Comment (npiyush97) [reply microsite/src/theme/customTheme.scss:9]: yes, I want maintainer's view on this according to A11Y guidelines links should be underlined to make them distinguishable from surrounding text by this lighthouse passes this issue as fixed, if you don't want this any other suggestions?

Comment (npiyush97) [reply microsite/src/theme/customTheme.scss:9]: because then we have to come up with colors for contrast above specified in the lighthouse which will affect backstage core font colors.

Comment (npiyush97) [reply microsite/src/theme/customTheme.scss:9]: @Rugvip how about this as an alternative?
#__docusaurus > .main-wrapper{
  a{
    text-decoration: underline;
  }
}

File: microsite/docusaurus.config.js
Comment (emmaindal) [inline microsite/docusaurus.config.js:208]: Since this link takes you back to home of the microsite (meaning its a functional image), I'd recommend describing the purpose a bit more. E.g. `Backstage Home`. See for example the [Example 1: Image used alone as a linked logo](https://www.w3.org/WAI/tutorials/images/functional/#example-1-image-used-alone-as-a-linked-logo)

File: microsite/src/theme/customTheme.scss
Comment (emmaindal) [reply microsite/src/theme/customTheme.scss:9]: Hello, yes I think we should definitely make links underline as that way the link will have a [distinct style](https://dequeuniversity.com/rules/axe/4.8/link-in-text-block) and therefore meet the criterias. If we do not have a distinct style we would work on improving the color contrast but links should look like links so lets move forward with this change. With this said - I believe its safe to avoid underline in the navigation so we can update this to underline links thats part of the text content. 

@npiyush97 I think your suggestion above moves us closer to the goal but it also underlines links in the sidebar so if you can pinpoint the links in only the text content that should be enough!

**Review by emmaindal — APPROVED** [review]
Thank you so much @npiyush97 for taking a look and making these changes - I've added a couple of comments!

**Review by emmaindal — CHANGES_REQUESTED** [review]
Wops - meant to request changes!

**Review by emmaindal — APPROVED** [review]
amazing, this looks good to me! I'll let the @backstage/maintainers review from their perspective too!

The only additional thing that would improve even further is if we could apply the underline to links in the blog section as well `/blog`, right now it only gets applied to the docs section of the microsite so the checks still fail on blog posts. Maybe the if you use the `markdown` class - it will apply to both? 

Approving to unblock!

**Comment by npiyush97** [comment]
> amazing, this looks good to me! I'll let the @backstage/maintainers review from their perspective too!
> 
> The only additional thing that would improve even further is if we could apply the underline to links in the blog section as well `/blog`, right now it only gets applied to the docs section of the microsite so the checks still fail on blog posts. Maybe the if you use the `markdown` class - it will apply to both?
> 
> Approving to unblock!

if you want I can add underline in the blog also with something like this
```
#__docusaurus {
  .row{
    .col
    {
     &--7{
      a {
        text-decoration: underline;
      } 
    }
    }
  }
}
```

**Comment by emmaindal** [comment]
@npiyush97 that looks like a good option, I'm a bit unfamiliar with the classes used for docusaurus and if that class is used anywhere else on the site. What we know at this point is that we want links within content to be underlined, looking at three pages (docs, blog and releases) which has content within the page - the `markdown` class seems to be a pattern it follows. Do you think we can use that? 
Screenshot 2024-01-05 at 14 32 41Screenshot 2024-01-05 at 14 32 27Screenshot 2024-01-05 at 14 32 05

**Comment by npiyush97** [comment]
@emmaindal can you open different issue for blog 
docs and releases comes in the same markdown this pull will fix for docs and releases
you want me to change?
.theme-doc-markdown => .theme-doc-markdown .markdown

**Comment by emmaindal** [comment]
yes we can open a separate issue for that, no worries. Let's get this in! Thanks again for great work. @backstage/maintainers can you take a look? Thank you!

**Review by benjdlambert — APPROVED** [review]

**Pull request merged**
